test/specs/token.test.mjs

```diff
@@ -0,0 +1,230 @@
+        ...options,
+      },
+      AO_LOADER_HANDLER_ENV,
+    );
+  }
+
+  /**
+   * Handlers:
+   * - info
+   * - total supply
+   * - balance
+   * - balances
+   * - transfer
+   
+   We do not implement:
+   * - mint
+   * - burn
+   *
+   */
+
+  it('should get the process info', async () => {
+    const result = await handle({
+      Tags: [{ name: 'Action', value: 'Info' }],
+    });
+    /**
+     * Check for:
+     * - Name (string)
+     * - Ticker (string)
+     * - Logo (arweave ID)
+     * - Denomination (number as string)
+     */
+
+    const tags = result.Messages[0].Tags;
+    const ref = tags.find((t) => t.name === 'X-Reference')?.value;
+    const name = tags.find((t) => t.name === 'Name')?.value;
+    const ticker = tags.find((t) => t.name === 'Ticker')?.value;
+    const logo = tags.find((t) => t.name === 'Logo')?.value;
+    const denomination = tags.find((t) => t.name === 'Denomination')?.value;
+
+    assert.strictEqual(typeof name, 'string');
+    assert.strictEqual(typeof ticker, 'string');
+    assert.strictEqual(typeof logo, 'string');
+    assert.strictEqual(typeof denomination, 'string');
+    assert.strictEqual(Number.isInteger(parseInt(denomination)), true);
+    assert(ref, 'Not a reply message');
+  });
+
+  it('should get the balance of an account', async () => {
+    /**
+     * Test with caller as:
+     * - Recipient
+     * - Target
+     * - From
+     * Test as:
+     * - Owner
+     * - Anonymous
+     *
+     * Check for:
+     * - Balance (string int)
+     * - Ticker (string)
+     * - Account (string equal to target address we are checking balance of)
+     * - Data (same as balance, both in type and qty)
+     */
+    const targetAccount = ''.padEnd(43, 'target-0-');
+    const targetAccountTags = [
+      // Should return 0
+      { name: 'Recipient', value: targetAccount },
+      { name: 'Target', value: targetAccount },
+      // happy path (owner calls for balance) should return 1
+      { name: 'Recipient', value: STUB_ADDRESS },
+      { name: 'Target', value: STUB_ADDRESS },
+      null, // "equivalent to "from"
+    ];
+
+    for (const targetAccountTag of targetAccountTags) {
+      const result = await handle({
+        Tags: [
+          { name: 'Action', value: 'Balance' },
+          ...(targetAccountTag ? [targetAccountTag] : []),
+        ],
+      });
+      const tags = result.Messages[0].Tags;
+      const ref = tags.find((t) => t.name === 'X-Reference')?.value;
+
+      const balanceData = result.Messages[0].Data;
+      const balanceTag = tags.find((t) => t.name === 'Balance')?.value;
+      const ticker = tags.find((t) => t.name === 'Ticker')?.value;
+      const account = tags.find((t) => t.name === 'Account')?.value;
+
+      assert.strictEqual(typeof balanceData, 'string');
+      assert.strictEqual(typeof balanceTag, 'string');
+      assert.strictEqual(balanceData, balanceTag);
+      assert.strictEqual(Number.isInteger(parseInt(balanceData)), true);
```

Comment:
Specific to ANTs - in the case of other tokens these would need to be modified to bint checking